File: 0189-rotate-array/0189-rotate-array.cpp
void rev(int st,int en, vector<int> &nums) {
    int low = st, high = en - 1;
    while(low < high) {
        swap(nums[low],nums[high]);
        low++,high--;
    }
}
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        k = k % n;
        rev(0,n,nums);
        rev(0,k,nums);
        rev(k,n,nums);
    }
};

File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rev(int low,int high,int[] nums) {
        int left = low;
        int right = high - 1;
        while(left < right) {
            int temp = nums[left];
            nums[left] = nums[right];
            nums[right] = temp;
            left++;
            right--;
        }
    }

    public void rotate(int[] nums, int k) {
        int n = nums.length;
        k = k % n;
        rev(0,n,nums);
        rev(0,k,nums);
        rev(k,n,nums);
    }
}